packages/abi/src/parser/specifications/v1/parser.ts

```diff
@@ -0,0 +1,76 @@
+import { FuelError } from '@fuel-ts/errors';
+
+import type { Abi } from '../../abi';
+
+import { mapAttribute } from './map-attribute';
+import { ResolvableType } from './resolvable-type';
+import { ResolvedType } from './resolved-type';
+import type {
+  AbiConfigurableV1,
+  AbiFunctionInputV1,
+  AbiFunctionV1,
+  AbiLoggedTypeV1,
+  AbiMessageTypeV1,
+  AbiSpecificationV1,
+} from './specification';
+
+export class AbiParserV1 {
+  static parse(abi: AbiSpecificationV1): Abi {
+    const resolvableTypes = abi.metadataTypes
+      .map((metadataType) => new ResolvableType(abi, metadataType.metadataTypeId, undefined))
+      .filter(
+        (x) => x.type !== 'struct std::vec::RawVec' && x.type !== 'struct std::bytes::RawBytes'
+      );
+
+    const types = abi.concreteTypes.map((ct) => {
+      const resolvableType = resolvableTypes.find((rt) => rt.metadataTypeId === ct.metadataTypeId);
+
+      const resolvedType = resolvableType
+        ? resolvableType.resolve(ct)
+        : new ResolvedType({ type: ct.type, typeId: ct.concreteTypeId });
+
+      return resolvedType.toAbiType();
+    });
+
+    const getType = (typeId: string | number) => {
```

Comment:
In what scenarios would `typeId` be a number? `concreteTypeId` is a string.

Reply:
be228d889be35fd4b9726efa46f6e5e03b62eef3

Reply:
be228d889be35fd4b9726efa46f6e5e03b62eef3